src/hook/index.tsx

```diff
@@ -90,11 +90,26 @@ export const useRemixForm = <T extends FieldValues>({
   // Submits the data to the server when form is valid
   const onSubmit = useMemo(
     () =>
-      (data: T, e: any, formEncType?: FormEncType, formMethod?: FormMethod) => {
+      (
+        data: T,
+        e: any,
+        formEncType?: FormEncType,
+        formMethod?: FormMethod,
+        formAction?: string,
+      ) => {
         setIsSubmittingNetwork(true);
         setIsSubmittedSuccessfully(true);
         const encType = submitConfig?.encType ?? formEncType;
         const method = submitConfig?.method ?? formMethod ?? "post";
+
+        let action = submitConfig?.action;
+        if (!action && formAction) {
+          const formActionUrl = new URL(formAction, window.location.origin);
+          action = formActionUrl.origin === window.location.origin
+            ? `${formActionUrl.pathname}${formActionUrl.search}`
+            : formActionUrl.href;
+        }
```

Comment:
I don't like this part, it's "too smart" in the sense that it does not act as you would expect it as it adds the search and also uses window to get the current location, this gets handled internally by Remix anyways, it's enough to just do:
```ts
const action = submitConfig?.action??formAction;
```
and then pass that into the submit as if the action is not defined then Remix will handle it and submit to the current route by itself. if you want to add search params this can be defined with either the action on the form or in the submit config manually, I don't want to make assumptions for the users

Reply:
Agreed, it's fixed now. Thanks for the review!